packages/react-components/react-charts-preview/library/src/components/CommonComponents/Popover.tsx

```diff
@@ -0,0 +1,247 @@
+import * as React from 'react';
+import { Popover, PopoverSurface } from '@fluentui/react-popover';
+import { mergeClasses } from '@griffel/react';
+import type { PositioningVirtualElement } from '@fluentui/react-positioning';
+import { tokens } from '@fluentui/react-theme';
+import { useId } from '@fluentui/react-utilities';
+import { getAccessibleDataObject, Points, pointTypes } from '../../utilities/index';
+import { convertToLocaleString } from '../../utilities/locale-util';
+import { Shape } from '../Legends/shape';
+import { usePopoverStyles_unstable } from './usePopoverStyles.styles';
+import { YValueHover } from './CartesianChart.types';
+import { LegendShape } from '../Legends/Legends.types';
+import { PopoverComponentProps } from './Popover.types';
+
+const PopoverComponent: React.FunctionComponent<PopoverComponentProps> = React.forwardRef<
+  HTMLDivElement,
+  PopoverComponentProps
+>((props, forwardedRef) => {
+  const virtualElement: PositioningVirtualElement = {
+    getBoundingClientRect: () => ({
+      top: props.clickPosition!.y,
+      left: props.clickPosition!.x,
+      right: props.clickPosition!.x,
+      bottom: props.clickPosition!.y,
+      x: props.clickPosition!.x,
+      y: props.clickPosition!.y,
+      width: 0,
+      height: 0,
+    }),
+  };
+  props = { ...props, ...props.customCallout?.customCalloutProps };
+  const classes = usePopoverStyles_unstable(props);
+  const Legend = props.xCalloutValue ? props.xCalloutValue : props.legend;
+  const YValue = props.yCalloutValue ? props.yCalloutValue : props.YValue;
+  return (
+    <div id={useId('callout')}>
+      <Popover
+        positioning={{ target: virtualElement, autoSize: 'always', offset: 20, coverTarget: false }}
+        open={props.isPopoverOpen}
+        inline
+      >
+        <PopoverSurface>
+          {/** Given custom callout, then it will render */}
+          {props.customCallout && props.customCallout.customizedCallout && props.customCallout.customizedCallout}
```

Comment:
Consider destructuring props with defaults for this case.

Reply:
@srmukher Could you address this comment

Reply:
all props by default are undefined (optional) and there are not any props which has a default value